### PR TITLE
PREAPPS-2378: Local Storage | Update local folders component - Delete (move to Local Trash)

### PR DESCRIPTION
**Pull request by madhavdhuppe**

### CODE REVIEW

File: src/schema/schema.ts [hunk 1]
```diff
@@ -190,8 +190,14 @@ export function createZimbraSchema(
 						}))
 			},
 			Mutation: {
-				action: (_, variables) => {
+				action: (_, variables, context) => {
```
Comment (yashrajpchavda) [inline src/schema/schema.ts]: please pass a default value for context as `context = {}`, might fail when we try to access the `local` from context.

File: src/schema/schema.ts [hunk 2]
```diff
@@ -190,8 +190,14 @@ export function createZimbraSchema(
...
 					const { type, ...rest } = variables;
+					const { local } = context;
+
+					if (local && type === 'FolderAction') {
```
Comment (yashrajpchavda) [inline src/schema/schema.ts]: please remove the `FolderAction` check from here and add it in the local client if necessary as we would have only one handler which caters to all the actions in the local client.

File: src/schema/schema.ts [hunk 3]
```diff
@@ -190,8 +190,14 @@ export function createZimbraSchema(
 						}))
 			},
 			Mutation: {
-				action: (_, variables) => {
+				action: (_, variables, context = {}) => {
 					const { type, ...rest } = variables;
+					const { local } = context;
```
Comment (pl12133) [inline src/schema/schema.ts]: See https://github.com/Zimbra/zm-api-js-client/pull/178/files#r283481421

File: src/schema/schema.ts [hunk 4]
```diff
@@ -196,8 +196,14 @@ export function createZimbraSchema(
...
+
+					if (local) {
```
Comment (itsmeprashant) [inline src/schema/schema.ts]: return local ? localStoreClient.action(type, rest as ActionOptions) : client.action(type, rest as ActionOptions);

File: src/schema/schema.ts [hunk 5]
```diff
@@ -196,8 +196,14 @@ export function createZimbraSchema(
 						}))
 			},
 			Mutation: {
-				action: (_, variables) => {
+				action: (_, variables, context = {}) => {
```
Comment (itsmeprashant) [inline src/schema/schema.ts]: ( _, { type, ...rest }, context = {} )

**Pull request merged**